src/internal/types/themeData.ts

```diff
@@ -0,0 +1,15 @@
+export type ThemeSaveState = {
+  history: any; // json object
+  index: number;
+  hash: string;
+};
+
+export type ThemeHistories = {
+  histories: History[];
+  index?: number;
+};
+
+export type History<D = any> = {
+  state: D;
+  id?: string;
```

Comment:
This shouldn't be optional for us. I know it is in the Puck History but that's because they changed it to be optional for the user. We want it always present.